pkg/controller/controlplane/valuesprovider.go

```diff
@@ -655,10 +682,20 @@ func getControlPlaneShootChartValues(
 		return nil, err
 	}
 
+	// get the ca.crt for caBundle of the snapshot-validation webhook
+	secret := &corev1.Secret{}
+	if err := client.Get(ctx, kutil.Key(cluster.ObjectMeta.Name, string(v1beta1constants.SecretNameCACluster)), secret); err != nil {
```

Comment:
Now going 1 more time over this handling I guess it would be more suitable to use the ControlPlane namespace rather than the Cluster name.

```suggestion
	if err := client.Get(ctx, kutil.Key(cp.Namespace, string(v1beta1constants.SecretNameCACluster)), secret); err != nil {
```